.github/workflows/ant.yaml

```diff
@@ -44,19 +44,81 @@ jobs:
       - run: yarn aos:build
       - run: yarn test
 
-  notify:
+  publish:
     runs-on: ubuntu-latest
+    environment: main
     needs: [unit, integration]
+    permissions:
+      contents: write # to be able to publish a GitHub release
+    outputs:
+      published_lua_code_id: ${{ steps.publish-lua-code.outputs.srcTxId }}
+      # lua_code: ${{ steps.publish-lua-code.outputs.luaSourceCode }}
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4.0.2
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - run: yarn --frozen-lockfile
+      - name: Publish Lua Code
+        id: publish-lua-code
+        env:
+          WALLET: ${{ secrets.WALLET }}
+          GITHUB_SHA: ${{ github.sha }}
+        run: |
+          # Run the publish command once and store the output in a variable
+          PUBLISH_OUTPUT=$(yarn aos:publish 2>&1)
+
+          # Extract the source transaction ID and Lua source code from the stored output
+          LUA_SOURCE_CODE_TX_ID=$(echo "$PUBLISH_OUTPUT" | grep -oE "publishedLuaTxId\.[a-zA-Z0-9_-]{43}" | cut -d'.' -f2)
+          # LUA_SOURCE_CODE=$(echo "$PUBLISH_OUTPUT" | grep -oP 'luaSourceCode{{{\K.*(?=}}})')
+
+          # Set the output variables for the workflow step
+          echo "::set-output name=srcTxId::$LUA_SOURCE_CODE_TX_ID"
+          # echo "::set-output name=luaSourceCode::$LUA_SOURCE_CODE"
+
+      - name: Create Tag for Lua ID
+        run: |
+          git tag "${{ steps.publish-lua-code.outputs.srcTxId }}"
+          git push --tags
```

Comment:
i have not used it before in actions, but looks like `gh` cli has some easy commands for release/tag creation tied specifically to a file

```
   - name: Install GitHub CLI
        run: |
          sudo apt-get update
          sudo apt-get install gh -y

      - name: Authenticate GitHub CLI
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # GitHub automatically provides this token
        run: gh auth login --with-token <<< "$GITHUB_TOKEN"

      - name: Create GitHub Release and upload file
        run: |
          gh release create ${{ github.ref_name }} ./path/to/file --title "${{ github.ref_name }}" --notes "Release for ${{ github.ref_name }}"
```

Reply:
we can save this for the release block - no need to add now